crypto/keyring_streaming.go

```diff
@@ -174,8 +177,6 @@ func (msg *PlainMessageReader) VerifySignature() (err error) {
 	if msg.verifyKeyRing != nil {
 		processSignatureExpiration(msg.details, msg.verifyTime)
 		err = verifyDetailsSignature(msg.details, msg.verifyKeyRing)
-	} else {
-		err = errors.New("gopenpgp: no verify keyring was provided before decryption")
```

Comment:
The API is such that if the caller sets the verification key ring to nil, the library should not do any verification. So I don't think we need an error here.